runtime/marshaler.go

```diff
@@ -16,7 +16,7 @@ type Marshaler interface {
 	// NewEncoder returns an Encoder which writes bytes sequence into "w".
 	NewEncoder(w io.Writer) Encoder
 	// ContentType returns the Content-Type which this marshaler is responsible for.
-	ContentType() string
+	ContentType(v interface{}) string
```

Comment:
I don't think we can change this without breaking backwards compatibility. Could we add a new interface with this new method and then type assert at marshalling time? That way we could support both new and old marshallers.

Reply:
Changed it back and assert it at marshalling time. What do you think of this implementation?

Reply:
I see also the test failing on the go mod file. Does it expect a new line at the end of the file?